### PR TITLE
WHO The Lux Foundation Library

### PR DESCRIPTION
**Pull request by Simisays**

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/the_lux_foundation_library.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Name:The Lux Foundation Library
+ManaCost:no cost
+Types:Plane The Library
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_lux_foundation_library.txt:3]: I think you need to add it to `CardType.multiWordTypes`

File: forge-core/src/main/java/forge/card/CardType.java [hunk 2]
```diff
@@ -45,7 +45,7 @@ public final class CardType implements Comparable<CardType>, CardTypeView {
     private static final long serialVersionUID = 4629853583167022151L;
 
     public static final CardTypeView EMPTY = new CardType(false);
-    private static final Set<String> multiWordTypes = ImmutableSet.of("Serra's Realm", "Bolas's Meditation Realm", "Dungeon Master");
+    private static final Set<String> multiWordTypes = ImmutableSet.of("Serra's Realm", "Bolas's Meditation Realm", "Dungeon Master","The Library");
```
Comment (Agetian) [inline forge-core/src/main/java/forge/card/CardType.java:48]: Style: should have a space after the comma: , "The Library"

File: forge-game/src/main/java/forge/game/card/CounterType.java [hunk 3]
```diff
@@ -20,7 +20,7 @@ public class CounterType implements Comparable<CounterType>, Serializable {
     // Rule 122.1b
     static ImmutableList<String> keywordCounter = ImmutableList.of(
             "Flying", "First Strike", "Double Strike", "Deathtouch", "Haste", "Hexproof",
-            "Indestructible", "Lifelink", "Menace", "Reach", "Trample", "Vigilance");
+            "Indestructible", "Lifelink", "Menace", "Reach", "Shadow","Trample", "Vigilance");
```
Comment (Agetian) [inline forge-game/src/main/java/forge/game/card/CounterType.java:23]: Style: there should be a space after the comma: "Shadow", "Trample"

File: forge-gui/res/cardsfolder/upcoming/the_lux_foundation_library.txt [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+S:Mode$ Continuous | EffectZone$ Command | Affected$ Player | SetMaxHandSize$ Unlimited | Description$ Players have no maximum hand size.
+T:Mode$ DamageDone | ValidSource$ Creature.YouCtrl | ValidTarget$ Player | CombatDamage$ True | OptionalDecider$ You | TriggerZones$ Command | Execute$ TrigDraw | TriggerDescription$ Whenever a creature you control deals combat damage to a player, you may draw a card.
+SVar:TrigDraw:DB$ Draw | NumCards$ 1
+T:Mode$ ChaosEnsues | TriggerZones$ Command | Execute$ RolledChaos | TriggerDescription$ Whenever chaos ensues, put a shadow counter on target creature. (A creature with shadow can block or be blocked by only creatures with shadow.)
+SVar:RolledChaos:DB$ PutCounter | CounterType$ Shadow | ValidTgts$ Creature
+DeckHas:Ability$Counters
+Oracle:Players have no maximum hand size.\nWhenever a creature you control deals combat damage to a player, you may draw a card.\nWhenever chaos ensues, put a shadow counter on target creature. (A creature with shadow can block or be blocked by only creatures with shadow.
```
Comment (Agetian) [inline forge-gui/res/cardsfolder/upcoming/the_lux_foundation_library.txt:10]: SVar:AIRollPlanarDieParams:Mode$ Always | HasCreatureInPlay$ True

**Review by Agetian — APPROVED** [review]

**Pull request merged**